app/services/price_policy_updater.rb

```diff
@@ -39,10 +40,27 @@ def assign_attributes
 
   def save
     ActiveRecord::Base.transaction do
-      @price_policies.all?(&:save) || raise(ActiveRecord::Rollback)
+      (@price_policies.all?(&:save) && save_product) || raise(ActiveRecord::Rollback)
     end
   end
 
+  def save_product
+    keys_from_built_rate_starts = []
+
+    rate_starts_params = permitted_product_params[:rate_starts_attributes].clone
+    rate_starts_params.values.each_with_index do |dr, index|
+      if dr[:min_duration].blank?
+        if dr[:id].present?
+          rate_starts_params["#{index}"].merge! ({ _destroy: '1' })
```

Comment:
Curious how the various validations work together - for example:
- can you have a rate start with no duration rates?
- if you have a rate start with duration rates, and you remove the rate start, what happens?

Reply:
> can you have a rate start with no duration rates?
Pushed this to when I add Duration rates. I was planning on validating that and returning an error. Not sure where I'd place it, though. Maybe in the Updater itself?

> if you have a rate start with duration rates, and you remove the rate start, what happens?
Should be destroyed

Reply:
@LeticiaErrandonea for the second case

> if you have a rate start with duration rates, and you remove the rate start, what happens?

I think we should either show an error or use JS to display a confirmation modal

Reply:
So if a user wants to delete the Step, they need to delete all duration rates? I think that may be a lot of work for the user, but I don't know how many price groups is common to have